tensorflow_datasets/scripts/cleanup/url_filename_recorder.py

```diff
@@ -19,34 +19,25 @@
 import pathlib
 import requests
 from absl import app
-from typing import Dict
+from typing import Dict, List
 from concurrent import futures
 
-import tensorflow as tf
 import tensorflow_datasets as tfds
 from tensorflow_datasets.core.download import checksums
 
 
-TFDS_PATH = tfds.core.utils.tfds_path()
+TFDS_PATH = tfds.core.utils.tfds_write_path()
 
 
-def _get_builder_checksum_path(name) -> pathlib.Path:
-  """Get path of DatasetBuilder dataset."""
-  # instead of glob can also use
-  # tfds.builder_cls(name).__dict__.__module__.split('.')[1:-1] to get path
-  path = (
-    tf.io.gfile.glob(os.path.join(
-      TFDS_PATH, '*', name, 'checksums.tsv'))
-    or
-    # since yesno dataset folder name is inconsistent with the
-    # dataset.builder's dataset name (yes_no)
-    tf.io.gfile.glob(os.path.join(
-      TFDS_PATH, '*', name.replace('_', ''), 'checksums.tsv'))
-  )
-  path = path[0]
-  if isinstance(path, str):
-    path = pathlib.Path(path)
-  return path
+def _collect_path_to_update() -> List[tfds.core.utils.ReadWritePath]:
+  """Collect checksums paths to update."""
+  url_checksums_paths = list(checksums._checksum_paths().values())
+
+  builder_checksums_path = []
+  for name in tfds.list_builders():
+    if tfds.builder_cls(name).url_infos != None:
```

Comment:
When the same function is used multiple times, it's best to cache it. Here `builder_cls = tfds.builder_cls(name)`